src/types/excelsheet.rs

```diff
@@ -15,11 +15,38 @@ use pyo3::{pyclass, pymethods, PyObject, Python};
 
 use crate::utils::arrow::record_batch_to_pybytes;
 
+pub(crate) enum Header {
+    None,
+    At(usize),
+    With(Vec<String>),
+}
+
+impl Header {
+    pub(crate) fn new(header_row: Option<usize>, column_names: Option<Vec<String>>) -> Self {
+        match column_names {
+            Some(headers) => Header::With(headers),
+            None => match header_row {
+                Some(row) => Header::At(row),
+                None => Header::None,
+            },
+        }
+    }
+
+    pub(crate) fn header_offset(&self) -> usize {
+        match self {
+            Header::At(index) => *index + 1,
```

Comment:
Hmm weird that index need to be dereferenced, since `usize` is `Copy`

Reply:
nice catch :+1: